doc/source/user/tutorial-ma.rst

```diff
@@ -0,0 +1,364 @@
+to plot a dashed line (using the ``'--'`` line style). See the
+`matplotlib <https://matplotlib.org/>`_ documentation for more info on this.
+
+.. code::
+
+    >>> import matplotlib.pyplot as plt
+    >>> plt.plot(dates, nbcases.T, '--')
+    >>> selected_dates = [0, 3, 11, 13]
+    >>> plt.xticks(selected_dates, dates[selected_dates])
+    >>> plt.title("COVID-19 cumulative cases from Jan 21 to Feb 3 2020")
+    
+.. plot:: user/plot_covid_1.py
+    :align: center
+    :include-source: 0
+
+.. note::
+
+   If you are executing the commands above in the IPython shell, it might be
+   necessary to use the command ``plt.show()`` to show the image window.
+
+The graph has a strange shape from January 24th to January 21st. It would be
+interesing to know where this data comes from. If we look at the ``locations``
+array we extracted from the ``.csv`` file, we can see that we have two columns,
+where the first would contain regions and the second would contain the name of
+the country. However, only the first few rows contain data for the the first
+column (province names in China). Following that, we only have country names. So
+it would make sense to group all the data from China into a single row. For
+this, we'll use the :func:`numpy.where` function to determine the indices of
+rows with data from China. Next, we'll use the :func:`numpy.sum` function to sum
+all the selected rows (``axis=0``)::
+
+    >>> china_total = nbcases[np.where(locations[:, 1] == 'China')].sum(axis=0)
+    array([  247,   288,   556,   817,   -22,   -22,   -15,   -10,    -9,
+              -7,    -4, 11820, 14410, 17237])
+
+Something's wrong with this data - we are not supposed to have negative values
+in a cumulative data set. What's going on?
+
+**Missing data**
+
+Looking at the data, here's what we find: there is a period with
+**missing data**::
+
+    >>> nbcases
+    array([[  258,   270,   375, ...,  7153,  9074, 11177],
+       [   14,    17,    26, ...,   520,   604,   683],
+       [   -1,     1,     1, ...,   422,   493,   566],
+       ...,
+       [   -1,    -1,    -1, ...,    -1,    -1,    -1],
+       [   -1,    -1,    -1, ...,    -1,    -1,    -1],
+       [   -1,    -1,    -1, ...,    -1,    -1,    -1]])
+
+All the ``-1`` values we are seeing come from :func:`numpy.genfromtxt`
+attempting to read missing data from the original ``.csv`` file. Obviously, we
+don't want to compute missing data as ``-1`` - we just want to skip this value
+so it doesn't interfere in our analysis. (Note that this is not a problem with
+:func:`numpy.genfromtxt`; in this particular case, substituting the missing
+value with `0` might have been fine, but we'll see later that this is far from
+a general solution). After importing the :mod:`numpy.ma` module, we'll create a
+new array, this time masking the invalid values::
+
+   >>> from numpy import ma
+   >>> nbcases_ma = ma.masked_values(nbcases, -1)
```

Comment:
I'd argue this sort of is a problem with the default behavior of genfromtxt, inserting ``-1`` silently is a dangerous default behavior.

Perhaps it would be better to use the `usemask=True` argument to `genfromtxt` instead?

Reply:
I see what you mean, and that would be the best practice, but in this case I chose to do it in parts because of pedagogical reasons. Showing the problem before you solve it is usually more motivating than just giving the right answer from the start :) I can add a note about the `usemask` parameter there.